libsolidity/formal/SymbolicVariables.h

```diff
@@ -154,15 +159,30 @@ class SymbolicFunctionVariable: public SymbolicVariable
 		EncodingContext& _context
 	);
 
-	Expression increaseIndex();
-	Expression operator()(std::vector<Expression> _arguments) const;
+	Expression currentValue(solidity::TypePointer const& _targetType = TypePointer{}) const override;
+
+	// Explicit request the function declaration.
+	Expression currentFunctionValue() const;
+
+	Expression valueAtIndex(int _index) const override;
+
+	// Explicit request the function declaration.
+	Expression functionValueAtIndex(int _index) const;
+
+	Expression resetIndex() override;
+	Expression increaseIndex() override;
+
+	Expression operator()(std::vector<Expression> _arguments) const override;
 
 private:
 	/// Creates a new function declaration.
 	void resetDeclaration();
 
 	/// Stores the current function declaration.
 	Expression m_declaration;
+
+	/// Abstract representation.
+	SymbolicIntVariable m_abstract;
```

Comment:
```suggestion
	SymbolicIntVariable m_abstract{
		TypeProvider::uint256(),
		TypeProvider::uint256(),
		m_uniqueName + "_abstract",
		m_context
	};
```
This should actually work and resolve the duplication in the constructors... (base class members are already initialized when member-initialization is executed, so you can refer to them here)

Reply:
hu and I thought the values have to be const to be initialized directly.. learned another thing

Reply:
Ah yes, that's a lot better!
Updated